Wie 18/sponsors page

Aur3lia opened the pull request:
Completed Sponsors Page.

- Added SponsorSection into components file for implementing sponsor sub-headings
- Fixed Jane Street logo typo
- Added SponsorRows function for adding the different sponsor images in each section
- Added mobile responsiveness

angewen commented on app/(pages)/sponsors/page.tsx:
```diff
@@ -1,15 +1,141 @@
+          sx={{ 
+            position: 'relative',
+            width: '100%',
+            paddingBottom: '20%',
```
try using pb as shorthand, it can be used outside of sx too `pb='20%'`

angewen commented on components/SponsorSection.tsx:
```diff
@@ -0,0 +1,25 @@
+          border: 0,
+          borderTop: '2px solid #93AED3',
+          mt: 2,
+          mb: 5,
```
mt and mb can be outside of sx

angewen commented on app/(pages)/sponsors/page.tsx:
```diff
@@ -1,15 +1,141 @@
-import { Stack, Typography } from '@mui/joy';
+import { Box, Stack, Typography } from '@mui/joy';
```
unused import Typography

angewen commented on app/(pages)/sponsors/page.tsx:
```diff
@@ -1,15 +1,141 @@
+        mb={10}
+        sx={{
+          width: {xs: '440px', sm: '475px', md: '500px', lg: '525px'},
+          mx: 'auto',
```
mx can be outside sx

angewen reviewed (changes requested):
The sponsor logos could be smaller, they look very large on desktop lol

This size would be gd with a larger margins on the sides: (I just zoomed out on the page to 65%)
<img width="1440" alt="image" src="https://github.com/UNSW-WIESoc/website/assets/119286195/e443dcba-abad-400e-a8a0-85111cea20c3">


For comparison this is the current size at 100% on desktop:
<img width="1439" alt="image" src="https://github.com/UNSW-WIESoc/website/assets/119286195/a5950bc7-24a6-4ae4-8b8f-78b5f3aa803a">

angewen commented on app/(pages)/sponsors/page.tsx:
```diff
@@ -1,15 +1,141 @@
+            spacing={10}
+            justifyContent="center"
+            alignItems="center"
+            sx= {{ width: { xs: '80%', sm: '65%', md: '85%', lg: '70%'} }}
```
Alot of code repetition for this Stack and its styling, perhaps make it a component

Aur3lia commented:
Made changes to this PR:
- Used HorizontalLine component in SponsorSection component
- Fixed UNSW logo ratio
- Changed the other sponsors' image size
- Used Grid instead of Box to format the sponsor rows

sophiac6410 reviewed (approved):
looks great!
